storage-multipath-autopart.ks.in

```diff
@@ -0,0 +1,56 @@
+#test name: storage-multipath-autopart
+
+%ksappend repos/default.ks
+%ksappend common/common_no_payload.ks
+%ksappend payload/default_packages.ks
+
+# Reboot the installed system.
+reboot
+
+# Validate also on the first boot.
+%ksappend validation/success_on_first_boot.ks
+
+%pre
+# Create a script that's used for testing during the installation in %post
+# as well as on the installed system during the first boot.
+# It reports to /root/RESULT as well as stdout due to the first boot testing approach.
+cat > /tmp/validation_script.sh << 'EOF'
+check_mntpoint_on_mpath() {
+    local mount_point="$1"
+
+    # Get the block device of the mount point
+    blk_device="$(findmnt -o SOURCE --noheadings "${mount_point}")"
+    # Any device in the device tree of blk_device should be a multipath device
+    lsblk --inverse --raw --output TYPE --noheadings "${blk_device}" | grep -q mpath
+    mpath_found=$?
+
+    # Report a failure if no mpath device was detected
+    if [ $mpath_found -ne 0 ]; then
```

Comment:
I think you can skip the `$?` bit and just have the pipe commands in the `if`, but it's not really something critical.
```
if lsblk --inverse --raw --output TYPE --noheadings "${blk_device}" | grep -q mpath ; then
```

Reply:
1) It would make the `if` line much longer.
2) IMO, this version with the `mpath_found` variable is easier to understand when reading the code.

These were my reasons to implement it this way.